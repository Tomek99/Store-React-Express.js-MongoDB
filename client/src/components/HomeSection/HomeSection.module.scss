.HomeSection {
    display: flex;
    flex-direction: column;
    position: relative;
    height: 88vh;

    .slider {
        width: 100%;
        height: 100%;
        position: absolute;
        opacity: 0;

        img {
            width: 100%;
            height: 100%;
            object-fit: cover;
            object-position: bottom;
        }
    }

    .activeAnim {
        opacity: 1;
    }

    .containerDots {
        display: flex;
        justify-content: center;
        gap: 10px;
        position: absolute;
        left: 25px;
        bottom: 20px;
        padding: 10px;
        z-index: 2;
        cursor: default;

        .spanDot {
            width: 7px;
            height: 7px;
            border-radius: 50%;
            background-color: #ccc;
            cursor: pointer;
        }

        .activeDot {
            background-color: var(--main-color);
            border: var(--border);
        }
    }
}

@media screen and (max-width: 450px) {
    .HomeSection {
        height: 89vh !important;
    }
}

@media screen and (max-width: 375px) {
    .HomeSection {
        height: 86vh !important;
    }
}
@media screen and (max-width: 320px) {
    .HomeSection {
        height: 82vh !important;
    }
}
