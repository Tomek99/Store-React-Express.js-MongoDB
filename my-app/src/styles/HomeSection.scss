div.HomeSection {
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    background-image: url('../images/home-img.jpeg');
    background-repeat: no-repeat;
    background-position: center;
    background-size: cover; 
    height: 100vh;

    div {
        display: flex;
        flex-direction: column;
        margin-top: 10%;
        margin-left: 10%;
        color: #fff;
        gap: 20px 0;
        
        header {
            text-transform: uppercase;
            font-weight: bold;
            font-size: 5rem;
        }

        p {
            font-size: 1.5rem;
            font-weight: lighter;
        }
        
        button {
           align-self: flex-start;
           background-color: var(--main-color);
           color: #fff;
           font-size: 1.7rem;
           padding: 12px 35px 12px 35px;
           cursor: pointer;
        }
        button:hover{
            letter-spacing: .1rem;
        }
    }
}