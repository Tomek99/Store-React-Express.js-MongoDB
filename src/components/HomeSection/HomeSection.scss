.HomeSection {
    display: flex;
    flex-direction: column;
    background-image: url("../../../public/images/home-img.jpeg");
    background-repeat: no-repeat;
    background-position: center;
    background-size: cover;
    height: 100vh;

    .textAreaHome {
        display: flex;
        flex-direction: column;
        margin-top: 10%;
        margin-left: 10%;
        color: #fff;
        gap: 20px 0;

        header {
            text-transform: uppercase;
            font-weight: bold;
            font-size: 5rem;
        }

        p {
            font-size: 1.5rem;
            font-weight: lighter;
        }

        a {
            align-self: flex-start;
            background-color: var(--main-color);
            color: #fff;
            font-size: 1.7rem;
            padding: 12px 35px 12px 35px;
            cursor: pointer;
        }
        a:hover {
            letter-spacing: 0.1rem;
        }
    }
}
@media screen and (max-width: 900px) {
    .HomeSection {
        justify-content: center;
        .textAreaHome {
            header {
                font-size: 4rem !important;
            }

            p {
            }
        }
    }
}
