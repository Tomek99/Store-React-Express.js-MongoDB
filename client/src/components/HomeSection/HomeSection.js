import { React, useState, useEffect } from "react";
import styles from "./HomeSection.module.scss";
import dataHomeSlider from "../../data/dataHomeSlider.json";
import BtnSlider from "./BtnSlider/BtnSlider";
import Content from "./Content/Content";
import CarouselDots from "../CarouselDots/CarouselDots";
import { Swiper, SwiperSlide } from "swiper/react";
// Import Swiper styles
import "swiper/css";
import "swiper/css/pagination";
import "swiper/swiper.min.css";
import { Pagination, Navigation } from "swiper";

function HomeSection() {
  return (
    <div className={styles.HomeSection}>
      <div className={styles.homeSectionCenter}>
        <Swiper
          pagination={true}
          modules={[Pagination, Navigation]}
          className={`${"mySwiper"} ${styles.paginationLeftSide}`}
          style={{ height: "600px", width: "100%" }}
        >
          {dataHomeSlider.map((obj, index) => {
            return (
              <SwiperSlide key={index}>
                <Content obj={obj} slideIndex={index} />
                <img src={obj.url} alt={obj.title}></img>
              </SwiperSlide>
            );
          })}
        </Swiper>
      </div>
    </div>
  );
}

export default HomeSection;
