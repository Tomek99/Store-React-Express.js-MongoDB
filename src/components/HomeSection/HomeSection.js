import React from 'react'
import '../../styles/HomeSection.scss'

function HomeSection() {
    return (
        <div className='HomeSection' id='homeSection'>
            <div className='textAreaHome'>
                <header>fresh coffee in <br />the morning</header>
                <p>Lorem ipsum Dolor Sit Amet Consectetur Adipisicing Elit. Sit,<br /> Consequuntur? Et Veritatis Velit Nesciunt Odio!</p>
                <button type='button'>Get It Now</button>
            </div>
        </div>
    )
}

export default HomeSection