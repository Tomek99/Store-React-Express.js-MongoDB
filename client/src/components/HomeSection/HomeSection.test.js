import { render } from "@testing-library/react";
import HomeSection from "./HomeSection";

test("should render HomeSection component", () => {
  const { container } = render(<HomeSection />);
  expect(container).toMatchSnapshot();
});
