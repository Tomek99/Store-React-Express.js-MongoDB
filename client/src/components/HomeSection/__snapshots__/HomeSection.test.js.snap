// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render HeadingThree component 1`] = `
<div>
  <div
    class="HomeSection"
  >
    <div
      class="HomeSection"
    >
      <div>
        <div
          class="slider activeAnim"
        >
          <div
            class="content"
          >
            <header>
              fresh coffee in the morning
            </header>
            <p>
              Lorem ipsum Dolor Sit Amet Consectetur Adipisicing Elit. Sit, Consequuntur? Et Veritatis Velit Nesciunt Odio!
            </p>
            <a
              href="#menuSection"
            >
              get it now
            </a>
          </div>
          <img
            alt="fresh coffee in the morning"
            src="/images/home-img1.jpeg"
          />
        </div>
        <div
          class="slider"
        >
          <div
            class="content"
          >
            <header>
              product of the month
            </header>
            <p>
              Caffè Crema Dolce
            </p>
            <a
              href="#menuSection"
            >
              get it now
            </a>
          </div>
          <img
            alt="product of the month"
            src="/images/home-img2.jpeg"
          />
        </div>
        <div
          class="slider"
        >
          <div
            class="content"
          >
            <header>
              beans ground instant coffee
            </header>
            <p />
            <a
              href="#menuSection"
            >
              Products
            </a>
          </div>
          <img
            alt="beans ground instant coffee"
            src="/images/home-img3.jpeg"
          />
        </div>
      </div>
      <div
        class="btnDiv"
      >
        <button
          class="btnArrow btnArrowLeft"
        >
          <svg
            color="#fff"
            fill="currentColor"
            height="40"
            stroke="currentColor"
            stroke-width="0"
            style="color: rgb(255, 255, 255);"
            viewBox="0 0 1024 1024"
            width="40"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M752.145 0c8.685 0 17.572 3.434 24.237 10.099 13.33 13.33 13.33 35.143 0 48.473L320.126 515.03l449.591 449.591c13.33 13.33 13.33 35.144 0 48.474-13.33 13.33-35.142 13.33-48.472 0L247.418 539.268c-13.33-13.33-13.33-35.144 0-48.474L727.91 10.1C734.575 3.435 743.46.002 752.146.002z"
            />
          </svg>
        </button>
      </div>
      <div
        class="btnDiv"
      >
        <button
          class="btnArrow btnArrowRight"
        >
          <svg
            color="#fff"
            fill="currentColor"
            height="40"
            stroke="currentColor"
            stroke-width="0"
            style="color: rgb(255, 255, 255);"
            viewBox="0 0 1024 1024"
            width="40"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M271.653 1023.192c-8.685 0-17.573-3.432-24.238-10.097-13.33-13.33-13.33-35.144 0-48.474L703.67 508.163 254.08 58.573c-13.33-13.331-13.33-35.145 0-48.475 13.33-13.33 35.143-13.33 48.473 0L776.38 483.925c13.33 13.33 13.33 35.143 0 48.473l-480.492 480.694c-6.665 6.665-15.551 10.099-24.236 10.099z"
            />
          </svg>
        </button>
      </div>
      <div
        class="CarouselDots"
      >
        <span
          class="spanDot activeDot"
          data-testid="span-element"
        />
        <span
          class="spanDot"
          data-testid="span-element"
        />
        <span
          class="spanDot"
          data-testid="span-element"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`should render HomeSection component 1`] = `
<div>
  <div
    class="HomeSection"
  >
    <div
      class="HomeSection"
    >
      <div>
        <div
          class="slider activeAnim"
        >
          <div
            class="content"
          >
            <header>
              fresh coffee in the morning
            </header>
            <p>
              Lorem ipsum Dolor Sit Amet Consectetur Adipisicing Elit. Sit, Consequuntur? Et Veritatis Velit Nesciunt Odio!
            </p>
            <a
              href="#menuSection"
            >
              get it now
            </a>
          </div>
          <img
            alt="fresh coffee in the morning"
            src="/images/home-img1.jpeg"
          />
        </div>
        <div
          class="slider"
        >
          <div
            class="content"
          >
            <header>
              product of the month
            </header>
            <p>
              Caffè Crema Dolce
            </p>
            <a
              href="#menuSection"
            >
              get it now
            </a>
          </div>
          <img
            alt="product of the month"
            src="/images/home-img2.jpeg"
          />
        </div>
        <div
          class="slider"
        >
          <div
            class="content"
          >
            <header>
              beans ground instant coffee
            </header>
            <p />
            <a
              href="#menuSection"
            >
              Products
            </a>
          </div>
          <img
            alt="beans ground instant coffee"
            src="/images/home-img3.jpeg"
          />
        </div>
      </div>
      <div
        class="btnDiv"
      >
        <button
          class="btnArrow btnArrowLeft"
        >
          <svg
            color="#fff"
            fill="currentColor"
            height="40"
            stroke="currentColor"
            stroke-width="0"
            style="color: rgb(255, 255, 255);"
            viewBox="0 0 1024 1024"
            width="40"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M752.145 0c8.685 0 17.572 3.434 24.237 10.099 13.33 13.33 13.33 35.143 0 48.473L320.126 515.03l449.591 449.591c13.33 13.33 13.33 35.144 0 48.474-13.33 13.33-35.142 13.33-48.472 0L247.418 539.268c-13.33-13.33-13.33-35.144 0-48.474L727.91 10.1C734.575 3.435 743.46.002 752.146.002z"
            />
          </svg>
        </button>
      </div>
      <div
        class="btnDiv"
      >
        <button
          class="btnArrow btnArrowRight"
        >
          <svg
            color="#fff"
            fill="currentColor"
            height="40"
            stroke="currentColor"
            stroke-width="0"
            style="color: rgb(255, 255, 255);"
            viewBox="0 0 1024 1024"
            width="40"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M271.653 1023.192c-8.685 0-17.573-3.432-24.238-10.097-13.33-13.33-13.33-35.144 0-48.474L703.67 508.163 254.08 58.573c-13.33-13.331-13.33-35.145 0-48.475 13.33-13.33 35.143-13.33 48.473 0L776.38 483.925c13.33 13.33 13.33 35.143 0 48.473l-480.492 480.694c-6.665 6.665-15.551 10.099-24.236 10.099z"
            />
          </svg>
        </button>
      </div>
      <div
        class="CarouselDots"
      >
        <span
          class="spanDot activeDot"
          data-testid="span-element"
        />
        <span
          class="spanDot"
          data-testid="span-element"
        />
        <span
          class="spanDot"
          data-testid="span-element"
        />
      </div>
    </div>
  </div>
</div>
`;
